MISRA.md

```diff
@@ -0,0 +1,40 @@
+# MISRA Compliance
+
+The FreeRTOS-Plus-TCP library files conform to the [MISRA C:2012](https://www.misra.org.uk/MISRAHome/MISRAC2012/tabid/196/Default.aspx)
+guidelines, with some noted exceptions. Compliance is checked with Coverity static analysis.
+Since the FreeRTOS-Plus-TCP library is a low level library and might need to interact with the network hardware directly, some MISRA violations as listed below are not "fixed".
+Deviations from the MISRA standard are listed below:
+
+### Flagged by Coverity
+| Deviation | Category | Justification |
+| :-: | :-: | :-: |
+| Rule 1.2 | Advisory | Language extenstions are being used in a portable manner for each specific compiler. |
+| Rule 8.7 | Advisory | Functions without an external linkage are being used in other projects. Thus, function declarations are kept. |
+| Rule 19.2 | Advisory | Unions are used to reduce the memory footprint of the stack. |
```

Comment:
`union` is also used to represent packet formats in this code - see definition of `ProtocolPacket_t`.